scripts/olean_comparison.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+getCacheSize () {
+  rm -rf .lake/build/lib/Mathlib
+  >&2 lake exe cache clean!
+  >&2 lake exe cache get
+  du --summarize --human-readable .lake
+}
+
+getCacheSize () {
+  rm -rf .lake/build/lib/Mathlib
+  >&2 lake exe cache clean!
+  >&2 lake exe cache get
+  du .lake/build/lib/Mathlib | sed "s=^=${1} =; s=\t= =g"
+}
+
+# Because the `lean-pr-testing-NNNN` branches use toolchains that are "updated in place"
+# the cache mechanism is unreliable, so we don't test it if we are on such a branch.
+if [[ ! $(cat lean-toolchain) =~ ^leanprover/lean4-pr-releases:pr-release-[0-9]+$ ]]; then
+  git fetch origin master --depth 1
```

Comment:
Am I reading correctly that this:
- builds the PR
- uploads it to cache
- deletes the built oleans
- fetches ones cached for master
- measures their size
- re-fetches the oleans from the PR
- measures their size

?

That seems a little wasteful compared to the following alternative:
- build the PR
- measure the size of oleans
- upload them to cache (and possibly don't if the measured size exceeds some threshold; 10G seems quite safe as mathlib is at 3.7G now)
- delete the built oleans
- fetch ones cached for master
- measure their size

In fact, the fetch-from-master step could also be skipped as one can already tell what the size there is by looking at the CI logs for the latest build on master.

Reply:
I would prefer postponing uploading the new cache until the olean sizes are within parameters.

Reply:
If I can find the master sizes somewhere, that would be great!

I think that the final oleans that are left should be the ones from the branch, since the later steps (building Archive, Counterexamples, Shake and Lint) use them.

Reply:
@Vtec234 where can I retrieve the information about the sizes of `master`'s oleans?

Reply:
Ok, renaming the .lake folder out of the way before downloading `master`'s oleans already reduced by half the time of the step: now it takes about 30 seconds.  If I did not have to download `master`'s oleans, the step would be almost instantaneous, since it is just a very easy text processing command, but already as is, I think that it is much better!

Reply:
> I would prefer postponing uploading the new cache until the olean sizes are within parameters. 

Matt, do you want this to be part of this PR, or of a follow up one?

Reply:
As long as it eventually happens, either is fine

Reply:
> @Vtec234 where can I retrieve the information about the sizes of `master`'s oleans?

The sizes can be retrieved from the olean cache on Azure, but it needs a bit of logic to figure out exactly which sizes to fetch. A command for that could be added to `lake exe cache`.

Or, you could output all the sizes in CI logs and then fetch them from there in other CI builds, but it would be quite a lot of output.

Reply:
After looking into how Cache works, the former option looks infeasible without larger changes. This is because the cache itself does not store an association between mathlib4 commits and the oleans that correspond to them. `cache commit` can do that, but it [is not currently ran](https://github.com/leanprover-community/mathlib4/blob/25040ca7ee3241a452efbd6b7ee2e64b56a85513/.github/workflows/build.yml.in#L171) in CI. If the command were re-enabled, it would become possible to list the oleans for a specific mathlib commit and add up their sizes. However, I don't understand what the 'integrate with put-unpacked' part is about.

Reply:
CC #9687 @Kha

Reply:
Ok, after some discussion with Mario the latter option is the only one that seems to make sense. But it's also an optimization, and you can probably get away with the current strategy of just downloading the cache.